diff --git a/UserProfile/AvatarClass.php b/UserProfile/AvatarClass.php
index 0e8b7ed..e7b9d30 100644
--- a/UserProfile/AvatarClass.php
+++ b/UserProfile/AvatarClass.php
@@ -66,6 +66,10 @@ class wAvatar {
 			'alt' => 'avatar',
 			'border' => '0',
 		);
+
+		$defaultParams['src'] = 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7'; // Disabling image feature
+		$defaultParams['style'] = 'border-width:0;display:none;';
+
 		$params = array_merge( $extraParams, $defaultParams );

 		return Html::element( 'img', $params, '' );
diff --git a/UserProfile/SpecialUpdateProfile.php b/UserProfile/SpecialUpdateProfile.php
index 3ef038c..28f7ca0 100644
--- a/UserProfile/SpecialUpdateProfile.php
+++ b/UserProfile/SpecialUpdateProfile.php
@@ -106,6 +106,8 @@ class SpecialUpdateProfile extends UnlistedSpecialPage {
 				}
 			}

+$can_create = ($user->isAllowed('createpage'))?true:$can_create; // Please also allow internal RBx
+
 			// Boo, go away!
 			if ( $can_create == false ) {
 				global $wgSupressPageTitle;
diff --git a/UserProfile/SpecialUploadAvatar.php b/UserProfile/SpecialUploadAvatar.php
index c428f19..665d779 100644
--- a/UserProfile/SpecialUploadAvatar.php
+++ b/UserProfile/SpecialUploadAvatar.php
@@ -216,7 +216,12 @@ class SpecialUploadAvatar extends SpecialUpload {
 			</table>
 			</form>' . "\n";

-		$output .= '</div>';
+    $output .= '</div>';
+
+// Disabling file uploads see https://github.com/webplatform/mediawiki/issues/19
+$output  = '<div class="nmbox languages" style="padding:10px;"><strong>Apologies</strong>';
+$output .= ' Profile image uploads are disabled. If you want the feature back, you <a ';
+$output .= 'href="https://github.com/webplatform/mediawiki/issues/19">can vote this issue up</a>.</div>';

 		return $output;
 	}
diff --git a/UserProfile/i18n/en.json b/UserProfile/i18n/en.json
index e7295a0..e786833 100644
--- a/UserProfile/i18n/en.json
+++ b/UserProfile/i18n/en.json
@@ -164,6 +164,7 @@
 	"user-profile-userlevels-link": "User levels",
 	"user-profile-create-threshold-title": "Create user profile",
 	"user-profile-create-threshold-edits": "{{PLURAL:$1|one edit|$1 edits}}",
+	"user-profile-create-threshold-bogus": "enough edits to create pages. Once you have made enough, you will be granted access to this feature.",
 	"user-profile-create-threshold-votes": "{{PLURAL:$1|one vote|$1 votes}}",
 	"user-profile-create-threshold-comments": "{{PLURAL:$1|one comment|$1 comments}}",
 	"user-profile-create-threshold-comment-score-plus": "{{PLURAL:$1|one thumbs up|$1 thumbs up}}",
